tests/test_model.py

```diff
@@ -370,6 +370,40 @@ def test_setup_region(model, demda, tmpdir):
     assert np.all(model.region["value"] == 210000039)  # basin id
 
 
+def test_model_write_geoms(tmpdir):
+    model = Model(root=str(tmpdir), mode="w")
+    bbox = box(*[4.221067, 51.949474, 4.471006, 52.073727])
+    geom = gpd.GeoDataFrame(geometry=[bbox], crs=4326)
+    geom.to_crs(epsg=28992, inplace=True)
+    fn_region = str(join(tmpdir, "region.gpkg"))
+    geom.to_file(fn_region, driver="GPKG")
+    model.data_catalog.from_dict(
+        {
+            "region": {
+                "path": fn_region,
+                "data_type": "GeoDataFrame",
+                "driver": "vector",
+            }
+        }
+    )
+    model.setup_region({"geom": "region"})
+    model.write_geoms(to_wgs84=True)
+    region_geom = gpd.read_file(str(join(tmpdir, "geoms/region.geojson")))
+    assert region_geom.crs.to_epsg() == 4326
+
+
+def test_model_set_geoms(tmpdir):
```

Comment:
not sure what this method is testing? can you add some comments to explain?

Reply:
Its testing if set_geoms converts the geom to the model crs if the geom crs is not the same as the model crs.

Reply:
check. I've added some comments to clarify.